pkg/openscap/openscap_test.go

```diff
@@ -1,230 +1,157 @@
-			t.Errorf("%s expected to cause error:\n%v\nBut got:\n%v", k, v.expectedError, err)
-		}
-		if !v.shouldFail && err != nil {
-			t.Errorf("%s expected to succeed but failed with %v", k, err)
-		}
-		if v.evalReport != nil {
-			if !v.evalReport(report) {
-				t.Errorf("%s expected to succesfully evaluate the report", k)
-			}
-		}
+	if !bytes.Equal(contents, expectedCVEContents) {
+		t.Errorf("expected downloaded file contents (%v bytes) to match cve conents (%v bytes)", len(contents), len(expectedCVEContents))
 	}
+}
 
-	for k, v := range map[string]struct {
-		mountPath string
-		image     *docker.Image
+func TestSetOscapChrootEnv(t *testing.T) {
+	imageMountPath := "."
+	for k, test := range map[string]struct {
+		imageID   string
+		imageArch string
 	}{
-		"mount path does not exist":     {"nosuchdir", &docker.Image{}},
-		"mount path is not a directory": {"openscap.go", &docker.Image{}},
-		"image is nil":                  {".", nil},
+		"short image id":   {imageID: "aaaa", imageArch: "x86_x64"},
+		"long image id":    {imageID: "aaaaaaaaabbbbbbbbbbbbbcccccccccdddddddddddd", imageArch: "x86_x64"},
+		"no arch provided": {imageID: "aaaaaaaaabbbbbbbbbbbbbcccccccccdddddddddddd", imageArch: ""},
 	} {
-		if _, _, err := tsSuccessMocks.Scan(v.mountPath, v.image); err == nil {
-			t.Errorf("%s did not fail", k)
+		image := &docker.Image{ID: test.imageID, Architecture: test.imageArch}
+		if err := setOscapChrootEnv(imageMountPath, image); err != nil {
+			t.Errorf("%s: failed to set chroot env: %v", k, err)
 		}
-	}
 
-}
+		imageArch := image.Architecture
+		if imageArch == "" {
+			imageArch = unknown
+		}
+		maxLen := imageShortIDLen
+		if imageNameLen := len(image.ID); imageNameLen < maxLen {
+			maxLen = imageNameLen
+		}
 
-func notEmptyValue(k, v string) error {
-	if len(v) == 0 {
-		return fmt.Errorf("the value should'nt be empty for key %s", k)
+		for k, expectedV := range map[string]string{
+			"OSCAP_PROBE_ROOT":         imageMountPath,
+			"OSCAP_PROBE_OS_VERSION":   linuxVersionPH,
+			"OSCAP_PROBE_ARCHITECTURE": imageArch,
+			"OSCAP_PROBE_OS_NAME":      linux,
+			"OSCAP_PROBE_PRIMARY_HOST_NAME": fmt.Sprintf("docker-image-%s",
+				image.ID[:maxLen]),
+		} {
+			actualV := os.Getenv(k)
+			defer os.Unsetenv(k)
+			if actualV != expectedV {
+				t.Errorf("%s: env was not set correctly: %s expected, got %s", k, expectedV, actualV)
+			}
+		}
 	}
-	return nil
 }
 
-func TestSetOscapChrootEnv(t *testing.T) {
-	oldSetVar := osSetEnv
-	defer func() { osSetEnv = oldSetVar }()
+func TestOscapCommand(t *testing.T) {
+	wd, err := test.GetCurrentFileDir()
+	if err != nil {
+		t.Errorf("unexpected error: %v", err)
+	}
 
-	okImage := docker.Image{}
-	okImage.Architecture = "x86_64"
-	okImage.ID = "12345678901234567890"
-	tsGoodImage := &defaultOSCAPScanner{image: &okImage, imageMountPath: "."}
+	fedoraCpePrefix := "oval:org.open-scap.cpe.fedora:def:"
+
+	var cpePrefix string
+	var dist int
+	//requires fedora or centos distro on host
```

Comment:
the purpose of this is to allow a little bit more integration-ish tests (for running on travis, for example). the test is skipped if not running on a  target OS.

i figured this change would be ok seeing that image inspector itself won't scan non-RHEL/centos/fedora images atm anyway.

it might be appropriate to move these tests to an integration package (see #65)